lib/servers/SockJSServer.js

```diff
@@ -57,8 +57,14 @@ module.exports = class SockJSServer extends BaseServer {
     connection.close();
   }
 
-  // f should return the resulting connection
+  // f should return the resulting connection and, optionally, the connection headers
   onConnection(f) {
-    this.socket.on('connection', f);
+    this.socket.on('connection', (connection) => {
+      f(connection, connection.headers);
+    });
+  }
+
+  onConnectionClose(connection, f) {
```

Comment:
Why we need this?

Reply:
@evilebottnawi As I said above, I want to not assume anything about the `connection` object. Probably most `connection` objects will work like `connection.on('close', f)`, which is the case with both `sockjs` and `ws`. But maybe some other library implements it like `connection.onclose = f`.